homeassistant/components/knx/cover.py

```diff
@@ -127,7 +126,7 @@ async def after_update_callback(self, device: XknxDevice) -> None:
     @property
     def device_class(self) -> str | None:
         """Return the class of this device, from component DEVICE_CLASSES."""
-        if self._device_class in DEVICE_CLASSES:
+        if self._device_class:
             return self._device_class
         if self._device.supports_angle:
```

Comment:
This is dead code ? you need to move line 129 after 132, to get the effect.

Reply:
No, 132 shall just be evaluated if no device class is set by the user - thus `None`